### PR TITLE
Webhooks UI design

### LINKED ISSUE #6414: [CAL-755] Editing/Adding Webhooks UI Design - Layouts, spacing & type issues.
A lot of issues, particularly around the ping test area where font sizes are all over the place.

**Currently:**

**Should be**

[Settings/ Developer/ Webhooks/Add - Cal - Live (Figma)](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=26%3A59917&t=3hNynQ6hBKZ73ols-1)

I'll add more spec shortly but if anyone can see the issues by comparing with the designs feel free to take on.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-755](https://linear.app/calcom/issue/CAL-755/editingadding-webhooks-ui-design-layouts-spacing-and-type-issues)</sub>

### PR DESCRIPTION
**Pull request by joe-shajan**
## What does this PR do?

Fixes #6414

These are the changes I made.

Untitled-2023-01-13-1455

In the Figma file, the font weight and colour look different visually but the values are the same. I need more clarification about wether I should change it or not.

And also all the other colours are also the same.

Currently, this is the heading, but in Figma the back arrow is outside. But when I try to change it, it will affect other places also. So should I go with it or keep it as it is?

image

## Type of change

- [x] UI update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx [hunk 1]
```diff
@@ -34,12 +30,24 @@ export default function WebhookTestDisclosure() {
...
+          <div className="flex items-center space-x-1">
+            <h3 className="font-sm self-center font-medium text-black">{t("webhook_response")}</h3>
+            {mutation.data && (
+              <Badge variant={mutation.data.ok ? "green" : "red"}>
+                {mutation.data.ok ? t("passed") : t("failed")}
+              </Badge>
+            )}
+          </div>
+        </div>
+        <div className="rounded-b-md bg-black p-4 text-xs text-white">
```
Comment (emrysal) [inline packages/features/webhooks/components/WebhookTestDisclosure.tsx:44]: Tagging @Jaibles - but looking at this it looks good, only thing I can spot is that the font used for "No data yet." should be the `SF Mono` font. This is just the Mac OSX default so you can add `font-mono` to achieve this.

**Review by emrysal — CHANGES_REQUESTED** [review]
Tagging @Jaibles - but looking at this it looks good, only thing I can spot is that the font used for "No data yet." should be the `SF Mono` font. This is just the Mac OSX default so you can add

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx (hunk 1 above)
Comment (joeshajan) [reply packages/features/webhooks/components/WebhookTestDisclosure.tsx]: Added @emrysal

**Comment by joeshajan** [comment]
Hey @emrysal can you check if this is correct?

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx [hunk 2]
```diff
@@ -15,15 +15,11 @@ export default function WebhookTestDisclosure() {
...
-          <h3 className="font-sm self-center font-medium text-black">{t("webhook_response")}</h3>
-          {mutation.data && (
-            <Badge variant={mutation.data.ok ? "green" : "red"}>
-              {mutation.data.ok ? t("passed") : t("failed")}
-            </Badge>
-          )}
+    <>
+      <div className="flex justify-between">
+        <div>
+          <p className="font-medium text-black">{t("webhook_test")}</p>
```
Comment (CarinaWolli) [inline packages/features/webhooks/components/WebhookTestDisclosure.tsx:21]:
```suggestion
          <p className="text-sm font-semibold leading-5">{t("webhook_test")}</p>
```

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx [hunk 3]
```diff
@@ -15,15 +15,11 @@ export default function WebhookTestDisclosure() {
...
+          <p className="font-sm mb-4 text-gray-600">{t("test_webhook")}</p>
```
Comment (CarinaWolli) [inline packages/features/webhooks/components/WebhookTestDisclosure.tsx:22]:
```suggestion
          <p className="mb-4 text-sm font-normal text-gray-500">{t("test_webhook")}</p>
```

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx [hunk 4]
```diff
@@ -34,12 +30,24 @@ export default function WebhookTestDisclosure() {
...
       </div>
-      <div className="p-4">
-        {!mutation.data && <em>{t("no_data_yet")}</em>}
-        {mutation.status === "success" && (
-          <div className="overflow-x-auto  text-gray-900">{JSON.stringify(mutation.data, null, 4)}</div>
-        )}
+      <div className="space-y-0 rounded-md border-0 border-neutral-200 bg-white sm:mx-0 md:border">
+        <div className="flex justify-between border-b p-4">
+          <div className="flex items-center space-x-1">
+            <h3 className="font-sm self-center font-medium text-black">{t("webhook_response")}</h3>
```
Comment (CarinaWolli) [inline packages/features/webhooks/components/WebhookTestDisclosure.tsx:36]:
```suggestion
            <h3 className="self-center text-sm font-semibold leading-4">{t("webhook_response")}</h3>
```

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx [hunk 5]
```diff
@@ -34,12 +30,24 @@ export default function WebhookTestDisclosure() {
           {t("ping_test")}
         </Button>
       </div>
-      <div className="p-4">
-        {!mutation.data && <em>{t("no_data_yet")}</em>}
-        {mutation.status === "success" && (
-          <div className="overflow-x-auto  text-gray-900">{JSON.stringify(mutation.data, null, 4)}</div>
-        )}
+      <div className="space-y-0 rounded-md border-0 border-neutral-200 bg-white sm:mx-0 md:border">
```
Comment (CarinaWolli) [inline packages/features/webhooks/components/WebhookTestDisclosure.tsx:33]:
```suggestion
      <div className="space-y-0 rounded-md border-0 border-neutral-200 sm:mx-0 md:border">
```

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx [hunk 6]
```diff
@@ -34,12 +30,24 @@ export default function WebhookTestDisclosure() {
...
+          <div className="flex items-center space-x-1">
+            <h3 className="font-sm self-center font-medium text-black">{t("webhook_response")}</h3>
+            {mutation.data && (
+              <Badge variant={mutation.data.ok ? "green" : "red"}>
+                {mutation.data.ok ? t("passed") : t("failed")}
+              </Badge>
+            )}
+          </div>
+        </div>
+        <div className="rounded-b-md bg-black p-4 font-mono text-xs text-white">
```
Comment (CarinaWolli) [inline packages/features/webhooks/components/WebhookTestDisclosure.tsx:44]:
```suggestion
        <div className="rounded-b-md bg-black p-4 font-mono text-[13px] leading-4 text-white">
```

File: packages/features/webhooks/components/WebhookForm.tsx [hunk 7]
```diff
@@ -263,7 +263,7 @@ const WebhookForm = (props: {
             {t("cancel")}
           </Button>
           <Button type="submit" loading={formMethods.formState.isSubmitting}>
-            {props?.webhook?.id ? t("save") : t("create")}
+            {props?.webhook?.id ? t("save") : t("create webhook")}
```
Comment (CarinaWolli) [inline packages/features/webhooks/components/WebhookForm.tsx:266]:
```suggestion
            {props?.webhook?.id ? t("save") : t("create_webhook")}
```

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
I found some design issues and added suggestions @joeshajan can you please take a look at that, make sure to use the [figma design](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=26%3A59917&t=KX8CnkyeoHyu5Vu8-0) to compare if all font sizes, weights and line heights are as they should. Thank you 🙏

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx (hunk 3 above)
Comment (joeshajan) [reply packages/features/webhooks/components/WebhookTestDisclosure.tsx:22]: Hey, @CarinaWolli in figma its 600.

But visually 500 looking similar. So what should I go with 600 or 500?

image

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx (hunk 5 above)
Comment (joeshajan) [reply packages/features/webhooks/components/WebhookTestDisclosure.tsx:33]: Sorry I didn't notice it. Thanks, @emrysal for pointing it out.

File: packages/features/webhooks/components/WebhookTestDisclosure.tsx (hunk 3 above)
Comment (CarinaWolli) [reply packages/features/webhooks/components/WebhookTestDisclosure.tsx:22]: @joeshajan Sorry that's right 👍 still `font-sm` should be `text-sm` here

Comment (joeshajan) [reply packages/features/webhooks/components/WebhookTestDisclosure.tsx:22]: Yeah sure

**Comment by joeshajan** [comment]
Hey, @CarinaWolli and @emrysal I'm closing this pull request as I messed up when merging the main branch.

So I'm closing this PR and raising a new one #6803 with all the changes mentioned here.

**Pull request closed without merging**